src/lib/layout/SubHeader.tsx

```diff
@@ -6,106 +6,150 @@ import {
 
-  const subHeaderMenu: SubHeaderMenuInfo[] = [
-    { name: "Overview", slug: "/", icon: "home" },
-    { name: "Transactions", slug: "/txs", icon: "file" },
-    { name: "Blocks", slug: "/blocks", icon: "block" },
-  ];
-
-  if (govConfig.enabled)
-    subHeaderMenu.push(
-      { name: "Validators", slug: "/validators", icon: "validator" },
-      { name: "Proposals", slug: "/proposals", icon: "proposal" }
-    );
-
-  if (wasmConfig.enabled)
-    subHeaderMenu.push(
-      { name: "Codes", slug: "/codes", icon: "code" },
-      { name: "Contracts", slug: "/contracts", icon: "contract-address" }
-    );
-
-  if (moveConfig.enabled)
-    subHeaderMenu.push({
-      name: "Modules",
-      slug: "/modules",
-      icon: "contract-address",
-    });
-
-  if (nftConfig.enabled)
-    subHeaderMenu.push({
-      name: "NFTs",
-      slug: "/nft-collections",
-      icon: "group",
-    });
-
-  if (poolConfig.enabled)
-    subHeaderMenu.push({ name: "Osmosis Pools", slug: "/pools", icon: "pool" });
-
   const isCurrentPage = useIsCurrentPage();
 
-  const activeColor = "primary.light";
-
-  const trackOnClick = (tab: string) => {
-    track(AmpEvent.USE_TOPBAR, { tab });
-  };
+  const subMenu: SubHeaderMenuInfo[] =
+    tier === "full"
+      ? [
+          { name: "Overview", slug: "/", icon: "home" },
+          { name: "Transactions", slug: "/txs", icon: "file" },
+          { name: "Blocks", slug: "/blocks", icon: "block" },
+          ...(govConfig.enabled
+            ? [
+                {
+                  name: "Validators",
+                  slug: "/validators",
+                  icon: "validator" as IconKeys,
+                },
+                {
+                  name: "Proposals",
+                  slug: "/proposals",
+                  icon: "proposal" as IconKeys,
+                },
+              ]
+            : []),
+          ...(wasmConfig.enabled
+            ? [
+                { name: "Codes", slug: "/codes", icon: "code" as IconKeys },
+                {
+                  name: "Contracts",
+                  slug: "/contracts",
+                  icon: "contract-address" as IconKeys,
+                },
+              ]
+            : []),
+          ...(moveConfig.enabled
+            ? [
+                {
+                  name: "Modules",
+                  slug: "/modules",
+                  icon: "contract-address" as IconKeys,
+                },
+              ]
+            : []),
+          ...(nftConfig.enabled
+            ? [
+                {
+                  name: "NFTs",
+                  slug: "/nft-collections",
+                  icon: "group" as IconKeys,
+                },
+              ]
+            : []),
+          ...(poolConfig.enabled
+            ? [
+                {
+                  name: "Osmosis Pools",
+                  slug: "/pools",
+                  icon: "pool" as IconKeys,
+                },
+              ]
+            : []),
+        ]
```

Comment:
Actually, I think to make it more easy to read, no need to destruct the array. You can just `.flat()` at the end outer array to make the nested array flatten instead.